prow/pubsub/subscriber/server.go

```diff
@@ -194,8 +199,16 @@ func (s *PullServer) handlePulls(ctx context.Context, projectSubscriptions confi
 					}
 					msg.ack()
 				})
-				if err != nil && !errors.Is(derivedCtx.Err(), context.Canceled) {
-					logrus.WithError(err).Errorf("failed to listen for subscription %s on project %s", sub.string(), project)
+				if err != nil {
+					if errors.Is(derivedCtx.Err(), context.Canceled) {
+						logger.WithError(err).Debug("Exiting as context cancelled")
+						return nil
+					}
+					if strings.Contains(err.Error(), "code = PermissionDenied") {
+						logger.WithError(err).Warn("Seems like missing permission.")
+						return nil
```

Comment:
I'm not familiar with how errGroup.Go works. When we return `nil`, does this mean that this topic will get re-tried again? Could you describe the before/after behavior more in your commit message or PR description?